sys/kern/src/arch/riscv32.rs

```diff
@@ -569,8 +567,24 @@ fn trap_handler(task: &mut task::Task) {
         Trap::Exception(Exception::InstructionFault) => unsafe {
             handle_fault(task, FaultInfo::IllegalText);
         },
+
         _ => {
-            panic!("Unimplemented exception");
+            cfg_if::cfg_if! {
+                if #[cfg(feature = "platform-interrupts")] {
```

Comment:
nit: I think the naming scheme around the `platform-interrupts` feature and the `platform_interrupt_handler` is a tad confusing.  The feature flag implies that external interrupts are not platform interrupts, while the function name implies they are.
Isn't the `platform_interrupt_handler` more of a `userspace_interrupt_handler`?  It just delegates any interrupt to any task that has registered it.

Reply:
It won't delegate any interrupt, only interrupts that are reserved for the platform. The feature flag is what enables tasks to request those irqs, hence the name.